server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java

```diff
@@ -277,6 +378,130 @@ private void checkIfUserCompactionCanceled() {
     }
   }
 
+  private void initializeSelection(
+      Map<ExternalCompactionId,ExternalCompactionMetadata> extCompactions, Tablet tablet,
+      Map<ExternalCompactionId,String> externalCompactionsToRemove) {
+    CompactionKind extKind = null;
+    boolean unexpectedExternal = false;
+    Set<StoredTabletFile> tmpSelectedFiles = null;
+    Boolean selAll = null;
+    Long cid = null;
+    Boolean propDel = null;
+    int count = 0;
+
+    ArrayList<String> reasons = new ArrayList<>();
+
+    for (Entry<ExternalCompactionId,ExternalCompactionMetadata> entry : extCompactions.entrySet()) {
```

Comment:
I made a comment on #2111 that this code should be refactored so that it could be unit tested.  Thinking it would be  best to wait until then to refactor it.

Reply:
Sure @milleruntime that would be helpful.  Was thinking those test could be done after this is merged but before release.  It would be easier to review on its own.